### ISSUE TITLE
Implement RotationalElements trait

### ISSUE THREAD

**Issue by AngusGMorrison** [lox-space/lox#10]
Copied from the original issue in Kerolox:

The RotationalElements trait should provide the coefficients for the implementation of the IAU body-fixed reference frames for celestial bodies.

The data can be extracted from a [PCK text kernel](https://github.com/lox-space/lox/blob/main/data/pck00011.tpc) from NASA NAIF. This is already implemented for other quantities and traits in the tools/lox_gen crate. The fields of interest are (with XXX being the NAIF id of the respective body):

BODYXXX_POLE_RA: Polynomial coefficients for the right ascension
BODYXXX_POLE_DEC: Polynomial coefficients for the declination
BODYXXX_PM: Polynomial coefficients for the prime meridian
BODYXXX_NUT_PREC_RA: Trigonometric polynomial coefficients for the right ascension
BODYXXX_NUT_PREC_DEC: Trigonometric polynomial coefficients for the declination
BODYXXX_NUT_PREC_PM: Trigonometric polynomial coefficients for the prime meridian
A potential trait implementation could look like this:

pub trait RotationalElements {
  fn right_ascension_coefficients() -> ?;
  fn declination_coefficients() -> ?;
  fn prime_meridian_coefficients() -> ?;
}
My idea would be to store the coefficients as associated constants of the respective trait implementations or let the code generator hard code the return values. I am not sure whether these arrays have the same lengths for all bodies so there may be the need for zero padding.

TL;DR: Extend the lox_gen code generator to generate traits as shown above for all celestial bodies for which these quantities are defined.

**Comment by helgee** [comment]
> Copied from the original issue in Kerolox:

Of course, this issue belongs in this repo 😬

**Comment by helgee** [comment]
We'll also need the nutation and precession coefficients which are not assigned to the body itself but to the barycenter of the respective satellite system, i.e. `BODY3_NUT_PREC_ANGLES` for the Earth-Moon system.

**Comment by AngusGMorrison** [comment]
@helgee There are fewer bodies with trigonometric data than there are with non-trig data. Under the approach in my draft PR, none of the bodies without trig data (the Sun, for example) would be able to implement `RotationalElements`, which seems overly restrictive.

What do you think about splitting these fields into two traits: `RotationalElements` and `TrigonometricRotationalElements`?
